app/components/channel_list/categories/header/index.tsx

```diff
@@ -20,20 +23,31 @@ const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
 }));
 
 type Props = {
-    heading: string;
+    category: CategoryModel;
+    hasChannels: boolean;
 }
 
-const CategoryHeader = (props: Props) => {
+const CategoryHeader = ({category, hasChannels}: Props) => {
     const theme = useTheme();
     const styles = getStyleSheet(theme);
 
+    // Hide favs if empty
+    if (!hasChannels && category.type === 'favorites') {
+        return (<></>);
+    }
+
     return (
         <View style={styles.container}>
             <Text style={styles.heading}>
-                {props.heading.toUpperCase()}
+                {category.displayName.toUpperCase()}
             </Text>
         </View>
     );
 };
 
-export default CategoryHeader;
+const withCategory = withObservables(['category'], ({category}: {category: CategoryModel}) => ({
```

Comment:
We are trying to move the `withX` functions to their own index file, separated from the componet (similar to what we do in webapp with redux). Would you mind doing that?

Reply:
Yup I've kept them separate; but I think it felt overkill for such a small component. Will stick to the convention though.